src/main/java/cs/f10/t1/nursetraverse/logic/commands/ExportCommand.java

```diff
@@ -101,4 +104,42 @@ private boolean indexesAllInBounds(Set<Index> indexes, int bound) {
         }
         return true;
     }
+
+    @Override
+    public boolean equals(Object that) {
+        if (this == that) {
+            return true;
+        } else if (that instanceof ExportCommand) {
+            return this.exportFileName.equals(((ExportCommand) that).exportFileName)
+                    && compareOptionalIndexSet(this.targetIndexes, ((ExportCommand) that).targetIndexes);
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Compares two optional sets of indexes and returns true if they have the same indexes.
+     */
+
+    private boolean compareOptionalIndexSet(Optional<Set<Index>> first, Optional<Set<Index>> second) {
```

Comment:
So apparently the reason `Set.equals()` (or even `Set.contains()`) doesn't work on its own is that it relies on the element class having overridden `hashCode()`, which `Index` has not done yet.

This should work fine:
```
@Override
public int hashCode() {
    return zeroBasedIndex;
}
```

Reply:
Tq for the changes, I have adopted them.